### PR TITLE
Add `kotlin.mpp.enableCompatibilityMetadataVariant=true`

### PR DESCRIPTION
**Pull request by JavierSegoviaCordoba**

### CODE REVIEW

**Review by nomisRev — APPROVED** [review]
Nice work @JavierSegoviaCordoba 👏 👏 
I think this is the best middle ground until all of this stabilises in the Kotlin eco-system.

As soon as Kotest adds the `kotlin.mpp.enableCompatibilityMetadataVariant` flag, or it becomes enabled by default we can remove this workaround. Yes?

**Comment by JavierSegoviaCordoba** [comment]
> Nice work @JavierSegoviaCordoba 👏 👏 I think this is the best middle ground until all of this stabilises in the Kotlin eco-system.
> 
> As soon as Kotest adds the `kotlin.mpp.enableCompatibilityMetadataVariant` flag, or it becomes enabled by default we can remove this workaround. Yes?

Yeah, we have to create a PR in Kotest if this works.

About the default, it will never be true because it is a flag to add support to how KMP was working in old Kotlin versions when Kotlin Multiplatform becomes stable all old libraries should be already using the new way (if they aren't abandoned).

**Review by raulraja — APPROVED** [review]

**Review by i-walker — APPROVED** [review]
thanks @JavierSegoviaCordoba !!

**Comment by JavierSegoviaCordoba** [comment]
Looks like disabling tests is not enough in order to avoid them being compiled when you run test tasks, so we can't run test tasks in GitHub Actions in order to be able to merge the PR (and later be able to create an alpha).

**Comment by nomisRev** [comment]
@JavierSegoviaCordoba can we remove the `sourceSets.test` completely 😅
That way it should not be included in the build, right?

**Comment by JavierSegoviaCordoba** [comment]
@nomisRev we are running windows specifying test source sets which probably fail. We have to change the workflow to avoid that too.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -29,13 +29,13 @@ jobs:
         uses: gradle/gradle-build-action@v2.2.0
         if: matrix.os != 'windows-latest'
         with:
-          arguments: --full-stacktrace build
+          arguments: --full-stacktrace assemble
```
Comment (i-walker) [inline .github/workflows/pull_request.yml:32]: are we still able to run tests in CI with this configuration?

Comment (JavierSegoviaCordoba) [reply .github/workflows/pull_request.yml:32]: No, the idea is publishing an alpha to try this one, and later when the alpha is available, revert the commit as we talked a few days ago

cc @nomisRev @raulraja

Comment (i-walker) [reply .github/workflows/pull_request.yml:32]: Perfect, thanks for clarifying.

**Comment by nomisRev** [comment]
We're still using `build` in some places @i-walker so this failed on `main`.

**Comment by i-walker** [comment]
I have the fix here https://github.com/arrow-kt/arrow/pull/2749

**Pull request merged**